### ISSUE TITLE
Haptic Learning : inferencing human features using deep networks

### ISSUE THREAD

**Issue by akshaybahadur21** [PyDataPune/Talks#17]
For providing haptic feedback, users have been dependent on external devices including buttons, dials, stylus or even touch screens. The advent of machine learning along with its integration with computer vision has enabled users to efficiently provide inputs and feedback to the system. A machine learning model consists of an algorithm which draws some meaningful correlation between data without being tightly coupled to a specific set of rules. It's crucial to explain the subtle nuances of the network also the use-case we are trying to solve. The main question, however, is to discuss the need to eliminate an external haptic system and use something which feels natural and inherent to the user.
To connect the dots, we will talk about the development of applications specifically aimed to localize and recognize human features which could then, in turn, be used to provide haptic feedback to the system.
These applications will range from recognizing digits, alphabets which the user can 'draw' at runtime; developing state of the art facial recognition system; predicting hand emojis along with Google's project of 'Quick, Draw' of hand doodles.
In this presentation, we will discuss the development of such applications using computer vision techniques to minimize CPU utilization. First, we will start with formulating and addressing a strong problem statement followed by a thorough literature review. Once these things are taken care of, we will discuss the data gathering part, followed by the algorithm evaluation and future scope. By the end of the session, the audience will have a clearer understanding of building vision based models along with emphasis to contribute to the open-source community.

**Comment by mayankskb** [comment]
Hey,

Thanks for making the issue. 
The topic and content seem interesting. Would you pen down the line up for the talk of topics and a flow.

Also, please let us know the time that you would need to execute the talk.

Thanks
Mayank Mishra

**Comment by akshaybahadur21** [comment]
@mayankskb 

In the beginning, I would address the crowd and convey my story of how I got into machine learning and data science. Followed by a quick demo of Digit Recognition. Then a walkthrough of how I built it and all the problems I faced as well as optimizing the solution.

Next, I would talk about my 2nd project Emojinator and how I came up with the idea. Some code walkthrough and future.

Finally, I would demonstrate my 3rd project Facial Recognition and the research paper behind the idea.

While giving each of the demos, I would be talking about the model used. How to get the data, and how to preprocess it. How and why to use transfer learning. Why is literature review the most important phase of your project? How contributing to the community helps you ultimately.



At the end of the session, the audience will have a clearer look at how to start building real-world projects on their own.

The audience will have a clearer understanding of how to proceed while building a model. They would be encouraged to read research paper and do some research of their own before proceeding with the project.

The emphasis would be to contribute to the data science opensource community.

**Comment by mayankskb** [comment]
Hey Akshay,

Thanks for making an issue. Presenting apologies for a delayed response. We would like to have your talk for July PyData Meetup. Will let you know about the dates shortly.

Keep your stuff ready.
Thanks 
Mayank Mishra

**Comment by akshaybahadur21** [comment]
@mayankskb - Thanks.
Looking forward to it.

**Comment by PyDataPune** [comment]
@akshaybahadur21 Once your content is ready can you let us know the time window required to execute it.

Thanks

**Comment by akshaybahadur21** [comment]
@PyDataPune - I am thinking of a 1 hour session with Q/A at the end.
The content is ready, I just need to arrange it in a python notebook for future usage.
When is the july meetup happening?

**Comment by PyDataPune** [comment]
Planning for July 27.

**Comment by akshaybahadur21** [comment]
Alright.
I will keep everything ready till then.

**Comment by akshaybahadur21** [comment]
@PyDataPune - Updated the topic.

**Comment by PyDataPune** [comment]
Cool.. Thanks Akshay!! Also do send me lines for your introduction.

Thanks

**Comment by akshaybahadur21** [comment]
@PyDataPune - 
Bio : My interest in computer science sparked when I was working on a women's safety application aimed towards the women welfare in India and since then I have been incessantly working on improving my skills. I stumbled upon machine learning then. I have made several open-source contributions in the field of ML and would continue to do so. I have made successful prototypes like an autonomous car, alphabet recognition, cancer classification, gesture recognition using learning models. These prototypes showcase the power of deep learning and how I can help your organization implement learning models to solve business cases. My ambition is to make valuable contribution towards the ML community and leave a message of perseverance and tenacity. Currently, I am working as a software engineer at Symantec, India. I am also deeply influenced by literature especially Shakespeare's work.

**Comment by akshaybahadur21** [comment]
@PyDataPune - I want to know how and where the session would be held.
Some details would be nice

**Comment by mayankskb** [comment]
Hey,

We are talking with Persistent to keep our meetup there and they are very well in sync with us.

We will be recording your session. So if you have slides/demo/paper,  keep it handy.

Also, you need to install screen recording software which can capture the presentation on your laptop.

If you have any other alternate go with that if not then please go through the following readme to get an available screen recorder for your OS:

https://github.com/sigmakappa/ScreenRecorders

Let me know if a screen recorder for your OS is not available here.

Also, after installing do testing of it that it is working properly or not. 

Thanks 
Mayank Mishra

**Comment by mayankskb** [comment]
In case of any other doubt feel free to write it. 

Thanks 
Mayank Mishra

**Comment by akshaybahadur21** [comment]
@mayankskb  - I have prepared a python notebook for the same.
For screen recording, I will be using an app for mac. Will let you know after testing.

**Comment by akshaybahadur21** [comment]
@mayankskb - Can you provide official confirmation to the event.
Thanks

**Comment by PyDataPune** [comment]
Yes, @akshaybahadur21 The event is confirmed and accessible via this link : https://www.meetup.com/PyData-Pune/events/262828324/ 
You need to do RSVP through this link and can also send it to anyone who is interested in this event.

Thanks

**Comment by akshaybahadur21** [comment]
@PyDataPune - Thankyou

**Comment by akshaybahadur21** [comment]
@PyDataPune, @mayankskb  - One suggestion. It would be great if you include Speaker name in the meetup link. That would be really helpful while sharing this.

**Comment by PyDataPune** [comment]
Yes!! That's a cool suggestion but this time we went out of word limit and hence need to remove a few things and even shortened the description. We are planning to circulate the introductory templates for speakers over our twitter handler lets see. When we can do this??? Probably will try to get this done by the weekend. 

@akshaybahadur21 Kindly do send us your contact details like mob number and email.. you can also email these things on pydata.pune@gmail.com. From there, someone from the admin team will pick it up and get in touch with you. 

Thanks
PyData Pune

**Comment by akshaybahadur21** [comment]
@PyDataPune - Thanks for having me.
Can you please share the recording and the next steps.

**Comment by mayankskb** [comment]
Hey,

We are processing the recording from the DSLR. Soon we will upload it in google drive and share the drive with you. So that you can do screen recording. 

Meanwhile please find the photographs of the event. I m sharing the photo album with you. 

Thanks 
Mayank Mishra

**Comment by akshaybahadur21** [comment]
@mayankskb - Thank you.
I will do the needful.
Thanks for the wonderful event.

**Comment by mayankskb** [comment]
Thanks, Akshay for your presence and the wonderful talk. We would like to have you more in the future. If you want to come to meetup any point in time ping me directly on WhatsApp 8380810361. Also, we would like to have more talks in the future so whenever you want to deliver feel free to create an issue and we will get back to you with the dates and timing... Now you also know the procedure so that would be helpful too

**Comment by akshaybahadur21** [comment]
@mayankskb - Sure.
Will let you know.

**Comment by mayankskb** [comment]
Closing this issue. Feel free to make a new issue for any new talk proposal.
